packages/amplify-category-function/src/index.ts

```diff
@@ -190,10 +193,30 @@ export const initEnv = async (context: $TSContext): Promise<void> => {
         resourceParamManager.setParams(s3Bucket);
         _.setWith(amplifyMeta, [categoryName, resourceName, 's3Bucket'], s3Bucket);
       }
-
-      // if the function has secrets, set the appId key in team-provider-info
-      if (getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: true }).length > 0) {
+    });
+  /**
+   * checking and updating if the lambda function resource needs tpi updated with secret Value
+   * all resources : all function resources in amplify cli project
+   */
+  allResources
+    .filter(resourceCategoryFilter)
+    .filter(resourceServiceFilter)
+    .forEach((r) => {
+      const { resourceName }: { resourceName: string } = r;
+      const resourceParamManager = envParamManager.getResourceParamManager(categoryName, resourceName);
+      // if the function has secrets, set the appId key in team-provider-info and checking local functions first
+      const localFunctionSecretNames = getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: false });
+      if (localFunctionSecretNames.length > 0) {
         resourceParamManager.setParam(secretsPathAmplifyAppIdKey, getAppId());
+      } else {
+        if (resourceParamManager.hasParam(secretsPathAmplifyAppIdKey)) {
+          resourceParamManager.deleteParam(secretsPathAmplifyAppIdKey);
+        }
+        const backendConfigFunctionKey = `AMPLIFY_function_${resourceName}_${secretsPathAmplifyAppIdKey}`;
+        // removing parameter in backend-config if present
+        if (getParametersControllerInstance().hasParameter(backendConfigFunctionKey)) {
+          getParametersControllerInstance().removeParameter(backendConfigFunctionKey);
+        }
```

Comment:
this should already be taken care of by the paramManager right? Is there some reason we need this to happen before save() is called on the paramManager?